rundetection/rules/loq_rules.py

```diff
@@ -0,0 +1,178 @@
+import xmltodict
+
+from rundetection.rules.rule import Rule
+
+if typing.TYPE_CHECKING:
+    from rundetection.job_requests import JobRequest
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class SansFileData:
+    title: str
+    type: str
+    run_number: str
+
+
+def _extract_run_number_from_filename(filename: str) -> str:
+    # Assume filename looks like so: LOQ00100002.nxs, then strip.
+    return filename.split(".")[0].lstrip("LOQ").lstrip("0")
+
+
+def _extract_cycle_from_file_path(file_path: Path) -> str:
+    # Assume path looks like so: /archive/NDXLOQ/Instrument/data/cycle_24_2/LOQ000001.nxs
+    return file_path.parts[-2]
+
+
+def _is_sample_transmission_file(sans_file: SansFileData, sample_title: str) -> bool:
+    return sample_title in sans_file.title and sans_file.type == "TRANS"
+
+
+def _is_sample_direct_file(sans_file: SansFileData) -> bool:
+    return ("direct" in sans_file.title.lower() or "empty" in sans_file.title.lower()) and sans_file.type == "TRANS"
+
+
+def _is_can_scatter_file(sans_file: SansFileData, can_title: str) -> bool:
+    return can_title == sans_file.title.split("_")[0] and sans_file.type == "SANS/TRANS"
+
+
+def _is_can_transmission_file(sans_file: SansFileData, can_title: str) -> bool:
+    return can_title in sans_file.title and sans_file.type == "TRANS"
+
+
+def _find_trans_file(sans_files: list[SansFileData], sample_title: str) -> SansFileData | None:
+    for sans_file in sans_files:
+        if _is_sample_transmission_file(sans_file=sans_file, sample_title=sample_title):
+            return sans_file
+    return None
+
+
+def _find_direct_file(sans_files: list[SansFileData]) -> SansFileData | None:
+    reversed_files = reversed(sans_files)
+    for sans_file in reversed_files:
+        if _is_sample_direct_file(sans_file=sans_file):
+            return sans_file
+    return None
+
+
+def _find_can_scatter_file(sans_files: list[SansFileData], can_title: str) -> SansFileData | None:
+    for sans_file in sans_files:
+        if _is_can_scatter_file(sans_file=sans_file, can_title=can_title):
+            return sans_file
+    return None
+
+
+def _find_can_trans_file(sans_files: list[SansFileData], can_title: str) -> SansFileData | None:
+    for sans_file in sans_files:
+        if _is_can_transmission_file(sans_file=sans_file, can_title=can_title):
+            return sans_file
+    return None
+
+
+def find_path_for_run_number(cycle_path: str, run_number: int) -> Path | None:
+    zeros = ""
+    potential_path = Path(f"{cycle_path}/LOQ{zeros}{run_number}.nxs")
+    # 10 is just a magic number, but we needed an unrealistic value for the maximum
+    while len(zeros) < 10 and not potential_path.exists():  # noqa: PLR2004
+        zeros += "0"
+        potential_path = Path(f"{cycle_path}/LOQ{zeros}{run_number}.nxs")
+    # 10 is just a magic number, but we needed an unrealistic value for the maximum
+    if len(zeros) == 10:  # noqa: PLR2004
+        return None
+    return potential_path
```

Comment:
Use `zfill()` for string manipulation and a `for` loop removes a lot of the duplicate code and the need for an incrementer

```suggestion
    for padding in range(1, 11):
        # 10 is the maximum allowable 0s to be added before None is returned
        potential_path = Path(f"{cycle_path}/LOQ{str(run_number).zfill(padding)}.nxs")
        if potential_path.exists():
            return potential_path
    return None
```

Reply:
Suggestion used :)